fxa/oauth.py

```diff
@@ -167,6 +168,17 @@ def authorize_code(self, sessionOrAssertion, scope=None, client_id=None,
         # This flow is designed for web-based redirects.
         # In order to get the code we must parse it from the redirect url.
         query_params = parse_qs(urlparse(resp["redirect"]).query)
+
+        # Make sure the redirect URL is authentic
+        if "state" not in query_params:
+            error_msg = "state missing in OAuth response"
+            raise OutOfProtocolError(error_msg)
```

Comment:
Yes, basically it was just to avoid the potential `KeyError` for the following test. I kind of always do this. So there is no particular reason behind this test and not that I have seen a response without that parameter in the URI.

Reply:
Sorry, to clarify: if we're going to check the state parameter, it's definitely worthwhile to have this additional check for whether it is present in the response. But *both* of the checks here could only fail if the server has done the wrong thing.